### PR TITLE
Remove package target framework

### PR DESCRIPTION
**Pull request by ericstj**
Removes the duplication between PackageTargetFramework and NuGetTargetMoniker.

Also replaces uses of PackageDestination with the more concise PackageTargetFramework list.

### CODE REVIEW

**Comment by ericstj** [comment]
Docs will come in a separate PR.  I wanted to get this in before it got stale.

File: src/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj [hunk 1]
```diff
@@ -9,29 +9,14 @@
...
-    <PackageDestination Include="lib/netstandard1.1">
-      <TargetFramework>netstandard1.1</TargetFramework>
-    </PackageDestination>
-    <!-- duplicate in portable folder so older nuget clients can consume -->
-    <PackageDestination Include="lib/portable-net45+win8+wpa81">
-      <TargetFramework>portable-net45+win8+wpa81</TargetFramework>
-    </PackageDestination>
-  </ItemGroup>
+    <PackageTargetFramework  Condition="'$(TargetGroup)' == ''">netstandard1.3;net46</PackageTargetFramework>
+    <!-- To allow this library to work on b4.5 runtimes and other old platforms
```
Comment (bartonjs) [inline src/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj]: "b4.5"?

**Comment by bartonjs** [comment]
I had originally expected the NuGetTargetMoniker to be the one that went away, becoming calculated from the other one.  Particularly since the format from PackageTargetFramework seems to be more aligned with what goes in project.json.

But I'll believe there's a good reason. (Given a "good reason" the change LGTM structurally)

File: src/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj (hunk 1 above)
Comment (bartonjs) [inline src/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj]: Ooh, I bet it was supposed to be a v.  That took a minute.

**Comment by ericstj** [comment]
"good reason": NuGetTargetMoniker is required by the public task which reads dependencies from project.json.  PackageTargetFramework is our own invention.  I changed build tools is to infer PackageTargetFramework from NuGetTargetMoniker (which is already required).

**Comment by ericstj** [comment]
@dotnet-bot test  Innerloop Windows_NT Release Build and Test

**Comment by weshaggard** [comment]
LGTM

**Pull request merged**